flocker/control/test/test_httpapi.py

```diff
@@ -1960,13 +1963,54 @@ def saved(ignored):
         'primary address, but the input schema currently requires the primary '
         'attribute. This will need to be addressed before or as part of '
         'https://clusterhq.atlassian.net/browse/FLOC-1404'
+        # Consider fixing this issue before implementing the resize API changes.
     )
 
+    def test_no_primary(self):
+        """
+        An update request without a primary address is allowed.
```

Comment:
This requirement might be better expressed in the `UpdateSizeDatasetTestsMixin`.  Those tests will presumably issue requests without `primary` in the body.  If this test is written here, what will it be posting?  An empty object?  I suppose we can make that work if we want... but it's not a particular useful operation I think. :smile: